tests/units/anta_tests/routing/test_bgp.py

```diff
@@ -1239,4 +1239,412 @@
+        "expected": {"result": "success"},
+    },
+    {
+        "name": "failure-no-routes",
+        "test": VerifyEVPNType2Route,
+        "eos_data": [{"vrf": "default", "routerId": "10.1.0.3", "asn": 65120, "evpnRoutes": {}}],
+        "inputs": {"vxlan_endpoints": [{"endpoint": "192.168.20.102", "vni": 10020}]},
+        "expected": {
+            "result": "failure",
+            "messages": ["The following VXLAN endpoint do not have any EVPN Type-2 route: [('192.168.20.102', 10020)]"],
+        },
+    },
+    {
+        "name": "failure-path-not-active",
+        "test": VerifyEVPNType2Route,
+        "eos_data": [
+            {
+                "vrf": "default",
+                "routerId": "10.1.0.3",
+                "asn": 65120,
+                "evpnRoutes": {
+                    "RD: 10.1.0.5:500 mac-ip 10020 aac1.ab4e.bec2 192.168.20.102": {
+                        "evpnRoutePaths": [
+                            {
+                                "routeType": {
+                                    "active": False,
+                                    "valid": True,
+                                },
+                            },
+                        ]
+                    },
+                },
+            },
+        ],
+        "inputs": {"vxlan_endpoints": [{"endpoint": "192.168.20.102", "vni": 10020}]},
+        "expected": {
+            "result": "failure",
+            "messages": [
+                "The following EVPN Type-2 routes do not have at least one valid and active path: ['RD: 10.1.0.5:500 mac-ip 10020 aac1.ab4e.bec2 192.168.20.102']"
+            ],
+        },
+    },
+    {
+        "name": "failure-multiple-routes-not-active",
+        "test": VerifyEVPNType2Route,
+        "eos_data": [
+            {
+                "vrf": "default",
+                "routerId": "10.1.0.3",
+                "asn": 65120,
+                "evpnRoutes": {
+                    "RD: 10.1.0.5:500 mac-ip 10020 aac1.ab4e.bec2 192.168.20.102": {
+                        "evpnRoutePaths": [
+                            {
+                                "routeType": {
+                                    "active": False,
+                                    "valid": True,
+                                },
+                            },
+                        ]
+                    },
+                    "RD: 10.1.0.6:500 mac-ip 10020 aac1.ab4e.bec2 192.168.20.102": {
+                        "evpnRoutePaths": [
+                            {
+                                "routeType": {
+                                    "active": False,
+                                    "valid": False,
+                                },
+                            },
+                        ]
+                    },
+                },
+            },
+        ],
+        "inputs": {"vxlan_endpoints": [{"endpoint": "192.168.20.102", "vni": 10020}]},
```

Comment:
can we add a case where one route have two paths and at least one path is active/valid then second route also have 2 path but none of them are active/valid

Reply:
Added in d74d13090dc1dca2ea38122e3a5749da7c3923c5